ArduPlane/GCS_Mavlink.cpp

```diff
@@ -242,7 +242,7 @@ void Plane::send_vfr_hud(mavlink_channel_t chan)
         (ahrs.yaw_sensor / 100) % 360,
         abs(throttle_percentage()),
         current_loc.alt / 100.0f,
-        barometer.get_climb_rate());
+        (soaring_controller.is_active() ? soaring_controller._displayed_vario_reading : barometer.get_climb_rate()));
```

Comment:
please make _displayed_vario_reading() private, and add a public get_vario_reading() as a public accessor